examples/classifier.c

```diff
@@ -15,6 +16,324 @@ float *get_regression_values(char **labels, int n)
+        }
+        time = what_time_is_it_now();
+
+        pthread_join(load_thread, 0);
+        train = buffer;
+        load_thread = load_data(args);
+
+        time = what_time_is_it_now();
+
+        float loss = 0;
+#ifdef GPU
+        if(ngpus == 1){
+            loss = train_network(net, train);
+        } else {
+            loss = train_networks(nets, ngpus, train, 4);
+        }
+#else
+        loss = train_network(net, train);
+#endif
+        printf("epoch: %d, batch: %ld, seen: %f, loss: %f, rate: %f, seconds: %lf, images: %ld, bad epochs: %d\n", epoch, get_current_batch(net), (float)(*net->seen)/N, loss,
+            get_current_rate(net), what_time_is_it_now()-time, *net->seen, bad_epochs);
+        free_data(train);
+
+        if(*net->seen/N > epoch) {
+            epoch = *net->seen/N;
+            update_epoch = 1;
+        }
+
+        if(update_epoch && (epoch % params.eval_epochs) == 0){
+            char buff[1024];
+            sprintf(buff, "%s/%s.weights",backup_directory, base);
+            save_weights(net, buff);
+            update_epoch = 0;
+
+            int cur_epoch = epoch / params.eval_epochs;
+
+            //train_accs[cur_epoch] = calculate_accurracy(datacfg, cfgfile, buff, "train");
+            //valid_accs[cur_epoch] = calculate_accurracy(datacfg, cfgfile, buff, "valid");
+
+            float** conf_train = calculate_confusion_matrix(datacfg, cfgfile, buff, "train");
+            float** conf_valid = calculate_confusion_matrix(datacfg, cfgfile, buff, "valid");
+
+            train_accs[cur_epoch] = eval(conf_train, classes);
+            valid_accs[cur_epoch] = eval(conf_valid, classes);
+
+            for(j = 0; j < classes; ++j)
+                free(conf_train[j]), free(conf_valid[j]);
+            free(conf_train), free(conf_valid);
+
+
+            printf("\n\nTrain metric: %f\nValidation metric: %f\n\n\n", train_accs[cur_epoch], valid_accs[cur_epoch]);
+
+            fprintf(training_log, "%f %f\n", train_accs[cur_epoch], valid_accs[cur_epoch]);
+
+            if(valid_accs[cur_epoch] > best_acc){
+                best_acc = valid_accs[cur_epoch];
+                sprintf(buff, "%s/%s-best.weights",backup_directory,base);
+                save_weights(net, buff);
+            }
+
+            if(cur_epoch > 0 && valid_accs[cur_epoch] > valid_accs[cur_epoch-1])
+                bad_epochs = 0;
+
+            else if(++bad_epochs > params.patience)
+                break;
+        }
+    }
+
+    char buff[1024];
+    sprintf(buff, "%s/%s.weights", backup_directory, base);
+    save_weights(net, buff);
+    pthread_join(load_thread, 0);
+
+    free_network(net);
+    if(labels) free_ptrs((void**)labels, classes);
+    free_ptrs((void**)paths, plist->size);
+    free_list(plist);
+    //free(valid_accs);
+    //free(train_accs);
+    fclose(training_log);
+    //free(base);
+}
+
+
+
+
+float** calculate_confusion_matrix (char *datacfg, char *filename, char *weightfile, char* set_name)
```

Comment:
To keep the consistency with the existing codebase, please rename the variable `filename` to `cfgfile`